javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/declarations/JavaParserAnonymousClassDeclaration.java

```diff
@@ -80,7 +72,8 @@ public JavaParserAnonymousClassDeclaration(ObjectCreationExpr wrappedNode,
         superTypeDeclaration =
                 JavaParserFactory.getContext(wrappedNode.getParentNode().get(), typeSolver)
                         .solveType(superTypeName)
-                        .getCorrespondingDeclaration();
+                        .getCorrespondingDeclaration()
+                        .orElseThrow(() -> new UnsolvedSymbolException("Unable to solve anonymous class parent"));
```

Comment:
Descriptive error messages are appreciated :+1: 

At the risk of being thick, when might this be thrown? 

My assumptions: 

1. Get the class or interface type of the anonymous class
2. Get the supertype (always `Object` or similar) -- should always exist, never a case where there is no super type.

... meaning that it happens when we've not been able to locate/solve the parent?

Depending on answers to the above, are we sure that throwing an exception within the constructor is what we want to do? (probably yes, just asking the question!)

Given that it's not a change in behaviour (if it's not solved, it would `throw new UnsupportedOperationException("CorrespondingDeclaration not available for unsolved symbol.");`), we can always deal with a TODO / a separate PR rather than this PR.

Reply:
> meaning that it happens when we've not been able to locate/solve the parent?

Exacly! That is an issue that was reported at https://github.com/javaparser/javaparser/issues/3045 and this change fixed it (https://github.com/javaparser/javaparser/pull/3044#issuecomment-764857331).

Should I move it to another PR?